doc/mgmt/gnmi/Config_Replace_gNOI.md

```diff
@@ -0,0 +1,226 @@
+# Config Replace Operation in gNOI <!-- omit in toc -->
+
+## Table of Content <!-- omit in toc --> 
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#abbreviations)
+- [1 Overview](#1-overview)
+- [2 Requirements](#2-requirements)
+  - [2.1 Functional Requirements](#21-functional-requirements)
+  - [2.2 CLI Requirements](#22-cli-requirements)
+- [3 Architecture Design](#3-architecture-design)
+- [4 High level design](#4-high-level-design)
+- [5 SAI API](#5-sai-api)
+- [6 Configuration and management](#6-configuration-and-management)
+  - [6.1 Manifest (if the feature is an Application Extension)](#61-manifest-(if-the-feature-is-an-application-extension))
+  - [6.2 CLI/YANG model Enhancements](#62-cli/yang-model-enhancements)
+  - [6.3 Config DB Enhancements](#63-config-db-enhancements)
+- [7 Warmboot and Fastboot Design Impact](#7-warmboot-and-fastboot-design-impact)
+- [8 Memory Consumption](#8-memory-consumption)
+- [9 Restrictions/Limitations](#9-restrictions/limitations)
+- [10 Testing Requirements/Design](#10-testing-requirements/design)
+  - [10.1 Unit Test cases](#101-unit-test-cases)
+  - [10.2 System Test cases](#102-system-test-cases)
+- [11 Open/Action items - if any](#11-open/action-items---if-any)
+ 
+## Revision
+
+| Rev | Date       | Author | Change Description |
+| --- | ---------- | ------ | ------------------ |
+| 0.1 | 03/21/2025 | Karuppiah Dharmaraj | Initial version    |
+
+
+## Scope
+
+This document outlines the high-level design of supporting Configuration Replacement via gNOI
```

Comment:
>gNOI [](http://example.com/codeflow?start=89&length=4)

gNOI is for operations, like "reboot". gNMI is for configuration management. Is it better to use gNMI protocol and gNMI client?

Reply:
The file copy operation will be defined as gNOI, enabling the config replace to be executed if a destination is set to the running-config.